sdk/core/core/src/az_span_private.h

```diff
@@ -35,6 +35,59 @@ _az_span_scan_until(az_span self, _az_predicate predicate, int32_t* out_index);
 
 AZ_NODISCARD az_result _az_is_expected_span(az_span* self, az_span expected);
 
+/**
+ * @brief Removes any white space from right and left of \p span. Function will create a new
+ * #az_span poiniting to the first non-white-space (` `, \\n, \\r, \\t) found in \p span and up to
+ * the last non-white-space char.
+ *
+ * @remarks If \p span is full of non-white-space characters, this function will retrun an az_span
+ * of size 0.
+ *
+ * Example:
+ *  az_span a = AZ_SPAN_FROM_STR("  text with   \\n spaces   ");
+ *  az_span b = _az_span_trim_white_space(a);
+ *  // assert( b ==  AZ_SPAN_FROM_STR("text with   \\n spaces"));
```

Comment:
How does this render in doxygen? I think it needs to be in a  code block to show up as `code`.
http://www.doxygen.nl/manual/commands.html#cmdcode